Update webpack dependency to v4

I'll work on fixing this last dependency (partitioned from original [Issue 95](https://github.com/webcompat/webcompat-reporter-extensions/issues/95)).

```
npm run build:firefox

> webcompat-reporter-addon@0.5.1 build:firefox /Users/mitaylor/dev/webcompat-reporter-extensions
> webpack --config firefox/webpack.config.js

The CLI moved into a separate package: webpack-cli.
Please install 'webpack-cli' in addition to webpack itself to use the CLI.
-> When using npm: npm install webpack-cli -D
-> When using yarn: yarn add webpack-cli -D
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! webcompat-reporter-addon@0.5.1 build:firefox: `webpack --config firefox/webpack.config.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the webcompat-reporter-addon@0.5.1 build:firefox script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mitaylor/.npm/_logs/2018-03-13T03_56_35_522Z-debug.log
mitaylor at omg-5 in ~/dev/webcompat-reporter-extensions on test/webpack/1*
🐓 npm install -D webpack-cli
```
ok, got that installed.

```
🐓 npm run build:firefox

> webcompat-reporter-addon@0.5.1 build:firefox /Users/mitaylor/dev/webcompat-reporter-extensions
> webpack --config firefox/webpack.config.js

Hash: 0b6b9f3c6bd237c9107d
Version: webpack 4.1.1
Time: 162ms
Built at: 3/12/2018 10:57:18 PM
                       Asset       Size  Chunks             Chunk Names
./dist/firefox/background.js   1.58 KiB       0  [emitted]  main
   ./dist/firefox/content.js  347 bytes          [emitted]  
./dist/firefox/manifest.json  799 bytes          [emitted]  
  ./dist/firefox/icon128.png   1.66 KiB          [emitted]  
   ./dist/firefox/icon32.png  906 bytes          [emitted]  
   ./dist/firefox/icon48.png   1.28 KiB          [emitted]  
   ./dist/firefox/icon64.png   1.78 KiB          [emitted]  
Entrypoint main = ./dist/firefox/background.js
   [0] ./firefox/addon.js + 1 modules 2.41 KiB {0} [built]
       | ./firefox/addon.js 350 bytes [built]
       | ./shared/base.js 2.07 KiB [built]
```

The following line surprises me a bit:

`Entrypoint main = ./dist/firefox/background.js`.

And yep, I see you're right, there's a nested `dist` directory:

```
🐓 ls dist/
dist
```

Looking at webpack.firefox.js:

```
module.exports = {
  entry: "./firefox/addon.js",
  output: {
    filename: "./dist/firefox/background.js"
  },
```

So yeah, something is goofy with the filepaths it would seem!

I just changed the `to` for the `CopyWebpackPlugin` section files to `../dist` instead of `./dist` and it seems like I made some progress:

<img width="333" alt="screen shot 2018-03-12 at 11 06 45 pm" src="https://user-images.githubusercontent.com/67283/37322267-8441751a-264a-11e8-824c-d218c4e674bd.png">

That might give you a hint how to fix it, or I see there's some debug options for the plugin that could help with path resolution: https://github.com/webpack-contrib/copy-webpack-plugin#options

Let me know if you get stuck! I'm on PTO until Thursday, but I'll log back online tomorrow night.

<img width="333" alt="screen shot 2018-03-12 at 11 06 45 pm" src="https://user-images.githubusercontent.com/67283/37322248-517526ae-264a-11e8-898a-0d53b0113bf2.png">


<img width="333" alt="screen shot 2018-03-12 at 11 06 45 pm" src="https://user-images.githubusercontent.com/67283/37322182-155eca9e-264a-11e8-8daa-4e76a2b49e1f.png">

--- a/chrome/webpack.config.js
+++ b/chrome/webpack.config.js
@@ -10,21 +10,21 @@ const version = require("../package.json").version;
 module.exports = {
   entry: "./chrome/addon.js",
   output: {
-    filename: "./dist/chrome/background.js"
+    filename: "../dist/chrome/background.js"
   },
   plugins: [
     new CopyWebpackPlugin([
       {
         from: "shared/content.js",
-        to: "./dist/chrome/content.js"
+        to: "../dist/chrome/content.js"
       },
       {
         from: "shared/*.png",
-        to: "./dist/chrome/[name].[ext]"
+        to: "../dist/chrome/[name].[ext]"
       },
       {
         from: "shared/manifest.json",
-        to: "./dist/chrome/[name].json",
+        to: "../dist/chrome/[name].json",
         transform: function(content, path) {
           let manifest = JSON.parse(content);
           // Derive addon versioning from package.json

--- a/firefox-mobile/webpack.config.js
+++ b/firefox-mobile/webpack.config.js
@@ -10,21 +10,21 @@ const version = require("../package.json").version;
 module.exports = {
   entry: "./firefox-mobile/addon.js",
   output: {
-    filename: "./dist/firefox-mobile/background.js"
+    filename: "../dist/firefox-mobile/background.js"
   },
   plugins: [
     new CopyWebpackPlugin([
       {
         from: "shared/content.js",
-        to: "./dist/firefox-mobile/content.js"
+        to: "../dist/firefox-mobile/content.js"
       },
       {
         from: "shared/*.png",
-        to: "./dist/firefox-mobile/[name].[ext]"
+        to: "../dist/firefox-mobile/[name].[ext]"
       },
       {
         from: "./firefox-mobile/manifest.json",
-        to: "./dist/firefox-mobile/manifest.json",
+        to: "../dist/firefox-mobile/manifest.json",
         transform: function(content, path) {
           // Add Firefox-specific bits to the manifest.json
           let manifest = JSON.parse(content);

--- a/firefox/webpack.config.js
+++ b/firefox/webpack.config.js
@@ -10,21 +10,21 @@ const version = require("../package.json").version;
 module.exports = {
   entry: "./firefox/addon.js",
   output: {
-    filename: "./dist/firefox/background.js"
+    filename: "../dist/firefox/background.js"
   },
   plugins: [
     new CopyWebpackPlugin([
       {
         from: "shared/content.js",
-        to: "./dist/firefox/content.js"
+        to: "../dist/firefox/content.js"
       },
       {
         from: "shared/*.png",
-        to: "./dist/firefox/[name].[ext]"
+        to: "../dist/firefox/[name].[ext]"
       },
       {
         from: "shared/manifest.json",
-        to: "./dist/firefox/[name].json",
+        to: "../dist/firefox/[name].json",
         transform: function(content, path) {
           let manifest = JSON.parse(content);
           // Derive addon versioning from package.json

--- a/opera/webpack.config.js
+++ b/opera/webpack.config.js
@@ -10,21 +10,21 @@ const version = require("../package.json").version;
 module.exports = {
   entry: "./opera/addon.js",
   output: {
-    filename: "./dist/opera/background.js"
+    filename: "../dist/opera/background.js"
   },
   plugins: [
     new CopyWebpackPlugin([
       {
         from: "shared/content.js",
-        to: "./dist/opera/content.js"
+        to: "../dist/opera/content.js"
       },
       {
         from: "shared/*.png",
-        to: "./dist/opera/[name].[ext]"
+        to: "../dist/opera/[name].[ext]"
       },
       {
         from: "shared/manifest.json",
-        to: "./dist/opera/[name].json",
+        to: "../dist/opera/[name].json",
         transform: function(content, path) {
           let manifest = JSON.parse(content);
           // Derive addon versioning from package.json

--- a/package.json
+++ b/package.json
@@ -37,7 +37,8 @@
     "shelljs": "^0.8.1",
     "shx": "^0.2.2",
     "web-ext": "^2.4.0",
-    "webpack": "^3.10.0"
+    "webpack": "^4.1.1",
+    "webpack-cli": "^2.0.10"
   },
   "license": "MPL-2.0"
 }

